### PR TITLE
Implement Postlogistics Cash on Delivery labels

### PR DESCRIPTION
**Pull request by guewen**
The webservice awaits the cash on delivery amount. When a whole sales order
is delivered in one picking / one package, the amount is the total of
the sales order. If it is split in several pickings / packages, we don't
know the value of each package and so we ask the user to fill the price on
the packages.

### CODE REVIEW

File: delivery_carrier_label_postlogistics/i18n/delivery_carrier_label_postlogistics.pot [hunk 1]
```diff
@@ -320,156 +326,146 @@ msgid "PDF"
...
+msgid "PostLogistics Franking License"
 msgstr ""
 
 #. module: delivery_carrier_label_postlogistics
-#: selection:delivery.carrier.template.option,postlogistics_type:0
-msgid "Additional Service"
+#: field:delivery.carrier,postlogistics_license_id:0
+#: field:postlogistics.service.group,postlogistics_license_ids:0
+#: field:res.company,postlogistics_license_ids:0
+msgid "PostLogistics Frankling License"
```
Comment (yvaucher) [inline delivery_carrier_label_postlogistics/i18n/delivery_carrier_label_postlogistics.pot]: There is an old typo I certainly did.

s/Frankling/Franking

File: delivery_carrier_label_postlogistics/i18n/delivery_carrier_label_postlogistics.pot [hunk 2]
```diff
@@ -320,156 +326,146 @@ msgid "PDF"
...
-msgid "Cancel"
+#: field:res.company,postlogistics_username:0
+msgid "Username"
 msgstr ""
 
 #. module: delivery_carrier_label_postlogistics
-#: view:postlogistics.config.settings:0
-msgid "Apply"
+#: view:postlogistics.config.settings:delivery_carrier_label_postlogistics.view_postlogistics_config_settings
+msgid "Web Service Authentification"
```
Comment (yvaucher) [inline delivery_carrier_label_postlogistics/i18n/delivery_carrier_label_postlogistics.pot]: An other former typo

s/Authentification/Authentication

File: delivery_carrier_label_postlogistics/i18n/fr.po [hunk 3]
```diff
@@ -18,348 +19,304 @@ msgstr ""
...
-#: field:res.company,postlogistics_username:0
-msgid "Username"
-msgstr ""
-
-#. module: delivery_carrier_label_postlogistics
-#: model:ir.model.fields,field_description:delivery_carrier_label_postlogistics.field_res_company_postlogistics_logo
-#: field:res.company,postlogistics_logo:0
-msgid "Company logo for PostLogistics"
+"Gives the sequence on company to define priority on license when multiple "
+"license are available for the same group of service."
```
Comment (yvaucher) [inline delivery_carrier_label_postlogistics/i18n/fr.po]: s/license/licenses

File: delivery_carrier_label_postlogistics/i18n/fr.po [hunk 4]
```diff
@@ -386,113 +342,174 @@ msgstr ""
...
+msgid "PostLogistics Franking License"
 msgstr ""
 
 #. module: delivery_carrier_label_postlogistics
-#: model:ir.model,name:delivery_carrier_label_postlogistics.model_delivery_carrier_template_option
-msgid "Delivery carrier template option"
+#: field:delivery.carrier,postlogistics_license_id:0
+#: field:postlogistics.service.group,postlogistics_license_ids:0
+#: field:res.company,postlogistics_license_ids:0
+msgid "PostLogistics Frankling License"
```
Comment (yvaucher) [inline delivery_carrier_label_postlogistics/i18n/fr.po]: typo

File: delivery_carrier_label_postlogistics/i18n/fr.po [hunk 5]
```diff
@@ -386,113 +342,174 @@ msgstr ""
...
+msgstr "Mettre à jour les services PostLogistics"
+
+#. module: delivery_carrier_label_postlogistics
+#: field:res.company,postlogistics_username:0
+msgid "Username"
+msgstr "Nom d'utilisateur"
+
+#. module: delivery_carrier_label_postlogistics
+#: view:postlogistics.config.settings:delivery_carrier_label_postlogistics.view_postlogistics_config_settings
+msgid "Web Service Authentification"
```
Comment (yvaucher) [inline delivery_carrier_label_postlogistics/i18n/fr.po]: typo from pot

File: delivery_carrier_label_postlogistics/stock.py [hunk 6]
```diff
@@ -29,6 +29,38 @@ class StockPicking(models.Model):
...
+                  'on the packages when a package contains products '
+                  'from different sales orders.')
+            )
+        order_moves = order.mapped('order_line.procurement_ids.move_ids')
+        picking_moves = self.move_lines
+        # check if the package delivers the whole sales order
+        if order_moves != picking_moves:
+            raise exceptions.Warning(
+                _('The cash on delivery amount must be manually specified '
+                  'on the packages when a sales orders is delivered '
```
Comment (yvaucher) [inline delivery_carrier_label_postlogistics/stock.py]: s/orders/order

**Comment by yvaucher** [comment]
:+1:

**Comment by guewen** [comment]
I'll squash the translations commits when I get a green build

**Comment by guewen** [comment]
Squashed commits

**Comment by guewen** [comment]
Rebased on 8.0

**Comment by jgrandguillaume** [comment]
:+1:

**Pull request merged**